modules/local/filter_bams.nf

```diff
@@ -0,0 +1,34 @@
+process FILTER_BAMS {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::samtools=1.16.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.15.1--h1170115_0' :
+        'quay.io/biocontainers/samtools:1.15.1--h1170115_0' }"
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path(bam), stdout  , emit: bam
+    path  "versions.yml"              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when 
+
+    script:
+    """
+    if [[ \$(samtools view -c $bam) -ge 1000 ]];
```

Comment:
I wanted to do a filtering based on the read numbers in the bam file using the samtools view command but I didn't figure out how to do this with the nf-core module. I kept getting errors with the input channel being in the wrong format and wasn't able to pass empty paths as input, because the nf-core module uses .fasta and .bai files as input, too. This is also why I ended up using a "custom" .nf script for the other use of samtools view where I filter the unique alignments from the bam file based on the qname field – I tried multiple ways to do this using the nf-core module but just couldn't make it work 😅 I thought this would be ok since there was another "custom" samtools view script that I used as a reference